osu.Game/Rulesets/Scoring/HealthProcessor.cs

```diff
@@ -31,16 +34,24 @@ public abstract partial class HealthProcessor : JudgementProcessor
         /// </summary>
         public bool HasFailed { get; private set; }
 
+        /// <summary>
+        /// List of triggers that caused fail
+        /// </summary>
+        public List<IHasFailCondition> FailTriggers = new List<IHasFailCondition>();
```

Comment:
Why is this a list? If it's because there can be more than one mod triggering failure at once and one of them has `RestartOnFail` while the other doesn't, then prioritise mods with `RestartOnFail = true` when deciding which mod triggered failure.

Also call this `ModTriggeringFailure` or something.

Reply:
as mentioned multiple mods can cause failure I thought storing them would prove useful